app/services/support_interface/interview_changes_export.rb

```diff
@@ -0,0 +1,93 @@
+module SupportInterface
+  class InterviewChangesExport
+    def data_for_export
+      interview_audits.map do |audit|
+        row_for_audit(audit)
+      end
+    end
+
+    def row_for_audit(audit)
+      interview = audit.auditable
+      application_choice = interview.application_choice
+      {
+        audit_id: audit.id,
+        audit_created_at: audit.created_at,
+        audit_type: audit.action,
+        interview_id: interview.id,
+        candidate_id: application_choice.candidate.id,
+        application_form_id: application_choice.application_form.id,
+        provider_code: application_choice.provider.code,
+        provider_user: audit_user(audit),
+        interview_status: interview_status(interview, audit),
+      }.merge(
+        interview_change(audit).transform_keys(&:to_sym),
+      ).merge(
+        {
+          interview_preferences: application_choice.application_form.interview_preferences,
+          application_submitted_at: application_choice.application_form.submitted_at,
+          course_code: application_choice.course.code,
+          course_location: application_choice.site.name_and_code,
+        },
+      )
+    end
+
+  private
+
+    def interview_audits
+      Audited::Audit.where(auditable_type: 'Interview')
+    end
+
+    def interview_status(interview, audit)
+      if audit.created_at < interview_time_when_change_made(interview, audit)
+        'upcoming'
+      else
+        'past'
+      end
+    end
+
+    def interview_time_when_change_made(interview, audit)
+      # We want to return the time the interview was scheduled for at the time of the audit.
+      # If the time was changed, then return the original time from the audited_changes.
+      # If the time was not changed, then it is either an upcoming interview (since the validation means you can't make it in the past)
+      # or it is a cancellation and the interview time on the interview model is representative.
+      interview_time_from_audit(audit).presence || interview.date_and_time
+    end
+
+    def interview_time_from_audit(audit)
+      audited_time = audit.audited_changes['date_and_time']
+      if audited_time.is_a?(Array)
+        Time.zone.parse(audited_time.first)
+      elsif audited_time.present?
+        Time.zone.parse(audited_time)
+      end
+    end
+
+    def changed_attributes(audit)
+      if audit.action == 'update'
+        audit.audited_changes.transform_values(&:last)
+      else
+        audit.audited_changes
+      end
+    end
+
+    def interview_change(audit)
+      changes = changed_attributes(audit)
+      {
+        'date_and_time' => '',
+        'cancelled_at' => '',
+        'cancellation_reason' => '',
+        'provider_id' => '',
```

Comment:
This is what is in the spec, since this is the column name in the interview model and what is actually being changed. Since we are interested in what has changed, I don't think we really need the code. This also avoids having to load the provider code by ID for each audit.

I'm open to changing it, but can't see a strong reason to yet

Reply:
Fair enough, as long as it's clear and there is documentation I think we can keep as is.